schemas/spec-v2.schema.json

```diff
@@ -76,6 +78,13 @@
             "default": "development",
             "enum": ["production", "development"]
         },
+        "integrationType": {
+            "type": "string",
+            "title": "Integration with Switchboard",
+            "description": "'none' means no integration; 'pull' means the tool pulls the data from Switchboard; 'native' means the Switchboard uses the tool api to send data to it",
+            "default": "pull",
+            "enum": ["none", "pull", "native"]
```

Comment:
Can you provide a bit more detail on goal for these values?
I though the idea was to differentiate the tools based on how the switchboard UI behaves for each type of tool. This would allow us to explicitly show first the tools where the switchboard UI can provide a fully integrated user experience.
From this UX point of view:
- "pull" and "native" seem to be the same
-  "none" seems to cover both: local tools and not integrated online tools

I see 3 user interface flow types when calling a tool:
- "Integrated" - can indeed be subdivided in "pull" and "native" if there is another reason to
- "Local" - the UI can try things like launch applications via browser URL schemes
- "External" - the UI cannot do more than present guiding information to the user about how to proceed

Reply:
First about the `manualParameters`:  a Switchboard tool has a list of parameters that can be filled in. Some of them are URL query parameters, and for them we have the `queryParameters` entry in the `webApplication` object. Others are URL path parameters, and for them we have the `pathParameters` entry. Similarly, the `manualParameters` are also parameters that the tool has, with the difference that they cannot be filled in by the Switchboard, they have to be manually set by the user. 

About the `integrationType`:  I am a bit confused about the purpose of this field. It cannot be used for the UI, since the information is already present elsewhere -- the algorithm is like this: if there's a `standaloneApplication` entry then it's a local application; if there's a `manualParameters` entry then it is like the `External` type you mentioned above; else it's a normal integrated application -- so I'm not sure we need it. It would be useful if we want to quickly check this field in queries, but we have to be aware that it can conflict with other entries.

Reply:
So the Switchboard cannot do much with `manualParameters` but provide textual information about them?

What about using a JSON schema conditional? we could have a parent element `Integration` with `integrationType` as child and `manualParameters` + `standaloneApplication` could be defined as a conditional based on the enum value of `integrationType`

Reply:
The `manualParameters` are similar to regular parameters in the sense that they are **matched**. A service is only recommended to the user if all the parameters are matched. And a service can have any combination of query/path/manual parameters -- for instance a language query parameter could be preselected and the user doesn't have to specify it, just to upload one manual parameter. 

The JSON conditional schema could work, of course.

Reply:
I think a conditional schema would be more clear. IMO it is better if we can define the data model (mainly) by the schema, and the implementation(s) follow(s) it.

Reply:
ok, let me try to make a suitable schema

Reply:
I updated the PR with a conditional schema, how's now?

Reply:
I like it. We add a bit of complexity, but in return we now have and "application type" specification defined by the schema alone. The Switchboard algorithm implements this specification.